build-support/bin/generate_github_workflows.py

```diff
@@ -946,10 +960,10 @@ def merge_ok(non_docs_only_jobs: list[str]) -> Jobs:
                 "steps": [
                     {
                         "run": dedent(
-                            """\
-                    merge_ok_docs_only="${{ needs.set_merge_ok_docs_only.outputs.merge_ok }}"
-                    merge_ok_not_docs_only="${{ needs.set_merge_ok_not_docs_only.outputs.merge_ok }}"
-                    if [[ "${merge_ok_docs_only}" == "true" || "${merge_ok_not_docs_only}" == "true" ]]; then
+                            f"""\
+                    merge_ok_docs_only="{gha_expr('needs.set_merge_ok_docs_only.outputs.merge_ok')}"
+                    merge_ok_not_docs_only="{gha_expr('needs.set_merge_ok_not_docs_only.outputs.merge_ok')}"
+                    if [[ "${{merge_ok_docs_only}}" == "true" || "${{merge_ok_not_docs_only}}" == "true" ]]; then
```

Comment:
Oh right, these are intended to be bash env var expansions ... confusing. So the space sandwhich is the difference in this case / that's why GH expressions have that syntax.

Reply:
Oh, right - it has nothing to do with the space sandwwich. The {{ -> { and so you get ${ vs ${{ in the end to disambiguate a GH expression from a bash env var expansion. Alrighty, I'm slow.

Reply:
Yep! I got confused as well in the transition, but since this is in a block of shell commands, I think coming to it with fresh eyes, without the context of the transition, makes it more obvious.

Reply:
Agreed.